### PR TITLE
Elements: Rules panel re-write

### PR DESCRIPTION
**Pull request by bvaughn**
### [Loom overview](https://www.loom.com/share/8cb11278bcf9498cb8fc415c9ff30635)

- [x] Rewrite Elements > Rules panel to improve performance
- [x] Delete old (no longer used) Rules components and CSS
  - There are still some old components because of the Layout and Computed tabs; maybe we should kill those too?
- [x] Update e2e test helper methods to use data attributes rather than CSS class names

### CODE REVIEW

File: src/devtools/client/inspector/markup/components/rules/RulesPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    deferredSelectedNodeId ?? undefined
+  );
+
+  const canHaveRules = !!node?.isElement;
+
+  const cachedStyles = cssRulesCache.read(
+    replayClient,
+    canHaveRules ? deferredPauseId : undefined,
+    canHaveRules ? deferredSelectedNodeId : undefined
+  );
```
Comment (bvaughn) [inline src/devtools/client/inspector/markup/components/rules/RulesPanel.tsx:45]: [This video](https://www.loom.com/share/b1168cda0b674f688a71e2b1a4b3e9f2) explains why I also made this change.

File: packages/e2e-tests/helpers/elements-panel.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { Locator, Page, expect } from "@playwright/test";
 import chalk from "chalk";
+import { string } from "yargs";
```
Comment (markerikson) [inline packages/e2e-tests/helpers/elements-panel.ts:3]: :hand: Accidental auto-import?

**Review by markerikson — APPROVED** [review]
One stray auto-import, but seems reasonable otherwise!

File: packages/e2e-tests/helpers/elements-panel.ts (hunk 2 above)
Comment (bvaughn) [reply packages/e2e-tests/helpers/elements-panel.ts:3]: Oops, y up

**Pull request merged**